### DISCUSSION TITLE
Why does adding a few newlines change the meaning of this `where`'s closure?

### DISCUSSION THREAD

**Discussion by MatrixManAtYrService** [nushell/nushell#12843] [Q&A] [answered]
Hello all.  I could use a second set of eyes here.  Thanks for taking a look.

This works like I want it to:

```nushell
{items : [{metadata: {name: foo, creationTimestamp: "2024-05-12T11:23:01Z"}, status: {conditions: [{reason: "PodCompleted"}]}}]}
  | get items
  | where { |it| ((date now) - ($it.metadata.creationTimestamp | into datetime) > 5min) and ($it.status.conditions | any { |cond| "reason" in $cond and $cond.reason == "PodCompleted"}) }
  | get metadata.name
```
```
╭───┬─────╮
│ 0 │ foo │
╰───┴─────╯
```

This fails:
```nushell
{items : [{metadata: {name: foo, creationTimestamp: "2024-05-12T11:23:01Z"}, status: {conditions: [{reason: "PodCompleted"}]}}]}
  | get items
  | where { |it| 
      ((date now) - ($it.metadata.creationTimestamp | into datetime) > 5min) 
      and ($it.status.conditions | any { |cond| "reason" in $cond and $cond.reason == "PodCompleted"})
    }
```
```
Error: nu::shell::external_command

  × External command failed
   ╭─[entry #105:4:1]
 4 │             ((date now) - ($it.metadata.creationTimestamp | into datetime) > 5min)
 5 │             and ($it.status.conditions | any { |cond| "reason" in $cond and $cond.reason == "PodCompleted"})
   ·                 ──────────────────────────────────────────────┬─────────────────────────────────────────────
   ·                                                               ╰── Cannot convert bool to a string
 6 │     }
   ╰────
  help: All arguments to an external command need to be string-compatible

  ```

I just added some newlines, why is it suddenly talking about an external command?

**Comment by ghost** [comment, marked as answer]
a few things i have noticed:

your `and` is probably missing braces on the right side

you can simplify the `any`. the following ones are equivalent:
```nu
| any { |cond| "reason" in $cond and $cond.reason == "PodCompleted" }
| any { |cond| $cond.reason? == "PodCompleted" }
```

you try to get `$it.metadata.name`, which does not exist (-> i have added a `| default testName` for my testing).

`where` is a bit special in nushell and does not require a closure (brackets are enough).

when applying all these things it works:
```nu
{items : [{metadata: {creationTimestamp: "2024-05-12T11:23:01Z"}, status: {conditions: [{reason: "PodCompleted"}]}}]}
| get items
| where (
  ((date now) - ($it.metadata.creationTimestamp | into datetime) > 6hr)
  and
  ($it.status.conditions | any {|cond| $cond.reason? == "PodCompleted" })
)
| each {|it| echo $it.metadata.name? | default testName }
```

as a tip for future debugging: remove code until it works and then readd code or add dummies until it breaks again: just removing one condition and the `and` gets it to work and adding `and true` broke it again indicating that the `and` is at fault.

**Reply by MatrixManAtYrService** [reply]
Thanks for getting back to me.  I apologize for making significant edits after posting, I may have introduced some confusion.

`$cond.reason?` is very helpful to know about, as is the business about `| default` thanks

Your code works:
```
where (
   ...$it...
   and ...$it...
 )
```
 
Mine doesn't:
```
 where { |it|
  ...$it...
  and ...$it...
}
```

I'll have to meditate on this until I can put accurate words to my mistake, but you've definitely helped me narrow it down, thank you.

**Comment by MatrixManAtYrService** [comment]
@Jan9103 helped me see what is going on here.  I ended up writing an issue for it: https://github.com/nushell/nushell/issues/12845
